components/server/src/bitbucket-server/bitbucket-server-api.ts

```diff
@@ -190,6 +191,50 @@ export class BitbucketServerApi {
         );
     }
 
+    async getBranchLatestCommit(
+        user: User,
+        params: {
+            repoKind: "projects" | "users" | string;
+            owner: string;
+            repositorySlug: string;
+            branch: string;
+        },
+    ): Promise<BitbucketServer.Branch | undefined> {
+        // @see https://developer.atlassian.com/server/bitbucket/rest/v811/api-group-repository/#api-api-latest-projects-projectkey-repos-repositoryslug-branches-get
+        // @see https://bitbucket.gitpod-dev.com/rest/api/1.0/users/huiwen/repos/mustard/branches?filterText=develop
+        const queryParam = qs.stringify({
```

Comment:
Is it expected that for invalid (not existing) branches the WS gets created on the default one? For GitHub it finds invalid branches
<img width="523" alt="image" src="https://github.com/gitpod-io/gitpod/assets/29888641/e76af671-b464-4c16-91ec-6ae347dcc958">

Reply:
Congrats on the number 500 btw 😆